File: backend/src/main/java/com/clinicmanagement/doctorleave/DoctorLeaveRequestService.java
package com.clinicmanagement.doctorleave;

import com.clinicmanagement.doctorleave.dto.DoctorLeaveRequestCreateRequest;
import com.clinicmanagement.doctorleave.dto.DoctorLeaveRequestResponse;
import com.clinicmanagement.doctorleave.dto.ReviewDoctorLeaveRequest;

import java.util.List;

public interface DoctorLeaveRequestService {

    DoctorLeaveRequestResponse createLeaveRequest(DoctorLeaveRequestCreateRequest request, String currentUserEmail);

    List<DoctorLeaveRequestResponse> getMyLeaveRequests(String currentUserEmail);

    void cancelLeaveRequest(Long id, String currentUserEmail);

    List<DoctorLeaveRequestResponse> getAllLeaveRequests();

    List<DoctorLeaveRequestResponse> getLeaveRequestsByStatus(DoctorLeaveRequest.LeaveStatus status);

    DoctorLeaveRequestResponse approveLeaveRequest(Long id, String adminEmail);

    DoctorLeaveRequestResponse rejectLeaveRequest(Long id, ReviewDoctorLeaveRequest request, String adminEmail);
}

File: frontend/src/services/doctorLeaveRequestService.js
import axiosClient from "./axiosClient.js";

const doctorLeaveRequestService = {
  // ── Doctor endpoints ────────────────────────────────────────────────────────

  /** Create a leave / schedule-change request */
  createLeaveRequest(data) {
    return axiosClient.post("/doctor/leave-requests", data);
  },

  /** Fetch the logged-in doctor's own requests */
  getMyLeaveRequests() {
    return axiosClient.get("/doctor/leave-requests/my");
  },

  /** Cancel a PENDING request (doctor-only) */
  cancelLeaveRequest(id) {
    return axiosClient.delete(`/doctor/leave-requests/${id}`);
  },

  // ── Admin endpoints ─────────────────────────────────────────────────────────

  /** Fetch all requests, optionally filtered by status */
  getAllLeaveRequests(status) {
    const params = status && status !== "ALL" ? { status } : {};
    return axiosClient.get("/admin/doctor-leave-requests", { params });
  },

  /** Approve a PENDING request */
  approveLeaveRequest(id) {
    return axiosClient.put(`/admin/doctor-leave-requests/${id}/approve`);
  },

  /** Reject a PENDING request — adminComment is required */
  rejectLeaveRequest(id, adminComment) {
    return axiosClient.put(`/admin/doctor-leave-requests/${id}/reject`, {
      adminComment,
    });
  },
};

export default doctorLeaveRequestService;
